src/libraries/System.Net.Security/src/System/Net/Security/SecureChannel.cs

```diff
@@ -954,7 +954,17 @@ internal bool VerifyRemoteCertificate(RemoteCertificateValidationCallback? remot
 
             try
             {
-                _remoteCertificate = CertificateValidationPal.GetRemoteCertificate(_securityContext, out remoteCertificateStore);
+                X509Certificate2? remoteCertificate = CertificateValidationPal.GetRemoteCertificate(_securityContext, out remoteCertificateStore);
```

Comment:
Nit: Can we call this `newRemoteCertificate` or something like that? It's confusing to have both `remoteCertificate` and `_remoteCertificate`

Reply:
updated. I use generic certificate name for the temp variable.